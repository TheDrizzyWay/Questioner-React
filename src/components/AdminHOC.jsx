import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import tokenExpired from '../utils/tokenExpired';

export const AdminHOC = ({ component: Component, ...rest }) => {
    const token = rest.auth.loggedIn ? rest.auth.loggedIn[0].token : null;
    const isadmin = rest.auth.loggedIn ? rest.auth.loggedIn[0].isadmin : null;
    return (<Route
        {...rest}
        render={() => {
            if (token && !tokenExpired(token) && isadmin) {
                return <Component />;
            }
            return <Redirect to="/signin" />;
        }}
    />);
};

const mapStateToProps = state => ({
    auth: state.auth
});

AdminHOC.propTypes = {
    auth: PropTypes.object,
    component: PropTypes.func
};

export default connect(mapStateToProps)(AdminHOC);
