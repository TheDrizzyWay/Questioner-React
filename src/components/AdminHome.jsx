import React, { useEffect } from 'react';
import { connect } from 'react-redux';
import Grid from '@material-ui/core/Grid';
import PropTypes from 'prop-types';
import MeetupCard from './MeetupCard';
import CreateMeetupModal from './modals/CreateMeetupModal';
import Spinner from './Spinner';
import Pagination from '../components/Pagination';
import * as actions from '../actions';

export const AdminHome = (props) => {
    useEffect(() => {
        props.getMeetups();
    }, []);

    const {
        isLoading, meetups, createMeetup,
        clearCreateError, createError, created,
        toggleCreateModal, showCreateModal
    } = props;

    return (
      <>
      <CreateMeetupModal
          createMeetup={createMeetup}
          createError={createError}
          clearCreateError={clearCreateError}
          created={created}
          toggleCreateModal={toggleCreateModal}
          showCreateModal={showCreateModal}
      />
        <section>
            {isLoading && <Spinner spinclass="center" />}
            <p>My meetups</p>
            <div className="meet_container">
                {meetups.paginatedResult ? (
                    <Grid container direction="row" spacing={0}
                        style={{ padding: 24, justifyContent: 'space-between' }}>
                        {meetups.paginatedResult.map(({ id, ...rest }) => (
                            <MeetupCard key={id} {...rest} />
                        ))}
                    </Grid>
                ) : (<div>'No meetups created yet.'</div>)}
            </div>
        </section>
        <Pagination />
      </>
    );
};

const mapStateToProps = state => ({
    meetups: state.meetups.meetups,
    isLoading: state.meetups.isLoading,
    createLoading: state.meetups.createLoading,
    createError: state.meetups.createError,
    created: state.meetups.created,
    showCreateModal: state.modals.showCreateModal
});

const mapDispatchToProps = {
    getMeetups: actions.getMeetups,
    createMeetup: actions.createMeetup,
    clearCreateError: actions.clearCreateError,
    toggleCreateModal: actions.toggleCreateModal
};

AdminHome.propTypes = {
    getMeetups: PropTypes.func.isRequired,
    meetups: PropTypes.oneOfType([PropTypes.object, PropTypes.array]),
    isLoading: PropTypes.bool,
    createMeetup: PropTypes.func,
    clearCreateError: PropTypes.func,
    createError: PropTypes.any,
    created: PropTypes.bool,
    toggleCreateModal: PropTypes.func,
    showCreateModal: PropTypes.bool
};

export default connect(mapStateToProps, mapDispatchToProps)(AdminHome);
