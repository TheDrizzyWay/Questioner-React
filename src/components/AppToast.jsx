import React, { Component } from 'react';
import Snackbar from '@material-ui/core/Snackbar';
import PropTypes from 'prop-types';

let openSnackbarFn;

class AppToast extends Component {
    componentDidMount () {
        openSnackbarFn = this.openSnackbar;
    }

  state = {
      open: false,
      message: ''
  };

  openSnackbar = ({ message }) => {
      const callback = this.props.callback ? this.props.callback : null;
      this.setState({
          open: true,
          message
      });
      setTimeout(() => {
          this.setState({
              open: false,
              message: ''
          });
          if (callback) callback();
      }, 3000);
  };

  render () {
      const message = (
          <span
              id="snackbar-message-id"
              dangerouslySetInnerHTML={{ __html: this.state.message }}
          />
      );

      return (
          <Snackbar
              anchorOrigin={{ vertical: 'top', horizontal: 'center' }}
              message={message}
              open={this.state.open}
          />
      );
  }
}

AppToast.propTypes = {
    callback: PropTypes.func
};

export const openSnackbar = ({ message }) => {
    openSnackbarFn({ message });
};

export default AppToast;
