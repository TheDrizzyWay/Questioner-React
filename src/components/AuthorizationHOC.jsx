import React from 'react';
import { Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import tokenExpired from '../utils/tokenExpired';

export const AuthorizationHOC = ({ component: Component, ...rest }) => {
    const token = rest.auth.loggedIn ? rest.auth.loggedIn[0].token : null;
    return (<Route
        {...rest}
        render={() => {
            if (token && !tokenExpired(token)) {
                return <Component />;
            }
            return <Redirect to="/signin" />;
        }}
    />);
};

const mapStateToProps = state => ({
    auth: state.auth
});

AuthorizationHOC.propTypes = {
    component: PropTypes.func,
    token: PropTypes.string
};

export default connect(mapStateToProps)(AuthorizationHOC);
