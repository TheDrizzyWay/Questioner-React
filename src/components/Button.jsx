import React from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';

const ButtonComponent = ({ name, onClick, type, disabled, className, children }) => {
    return (
        <Button
            type={type}
            name={name}
            disabled={disabled}
            onClick={onClick}
            className={className}
        >
            {children}
        </Button>
    );
};

export default ButtonComponent;
