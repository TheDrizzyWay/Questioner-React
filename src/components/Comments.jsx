import React from 'react';
import Pagination from './Pagination';
import createicon from '../assets/images/create-new.png';

const Comments = (props) => {
    return (
    <>
    <div className="add">
        <button id="modal-button"><img src={createicon} alt="plus icon" /></button>
        <p>Add a comment</p>
    </div>
    <main>
        <p id="topic-span">Topic 001</p>
        <p id="question-span" />
        <div className="container">
            <div className="questions">
                <p><span>Comment: </span>{props.comment}</p>
                <p><span>Posted By: </span>{props.postedby} <span> On </span>{props.newDate}</p>
            </div>
            <Pagination />
        </div>
    </main>
    <div>
        <a className="home" id="back">Back</a>
    </div>
    </>
    );
};

export default Comments;
