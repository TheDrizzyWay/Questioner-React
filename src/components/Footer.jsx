import React from 'react';

const Footer = () => {
    return (
        <footer>
            <div className="footer" />
        </footer>
    );
};

export default Footer;
