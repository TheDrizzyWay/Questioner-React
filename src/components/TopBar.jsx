import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';

const styles = {
    root: {
        flexGrow: 1
    },
    grow: {
        flexGrow: 1
    }
};

export const TopBar = (props) => {
    const { classes, section } = props;
    const handleLogout = () => {
        localStorage.clear();
        window.location.reload();
    };

    return (
        <div className={classes.root}>
            <AppBar position="static">
                <Toolbar>
                    <Typography variant="h6" color="inherit" className={classes.grow}>
                        {section}
                    </Typography>
                    <Button color="inherit" onClick={handleLogout}>Logout</Button>
                </Toolbar>
            </AppBar>
        </div>
    );
};

TopBar.propTypes = {
    classes: PropTypes.object.isRequired,
    section: PropTypes.string.isRequired
};

export default withStyles(styles)(TopBar);
