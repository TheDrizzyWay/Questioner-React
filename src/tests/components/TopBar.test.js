import React from 'react';
import { shallow } from 'enzyme';
import { TopBar } from '../../components/TopBar';

delete window.location;
window.location = { reload: jest.fn() };
const clearSpy = jest.spyOn(Object.getPrototypeOf(window.localStorage), 'clear');

const props = {
    section: 'section',
    classes: {
        sample: 'sample'
    }
};

let wrapper;

describe('<TopBar />', () => {
    it('Should render component successfully', () => {
        wrapper = shallow(<TopBar {...props} />);
        expect(wrapper.length).toEqual(1);
    });

    test('Should maintain existing snapshot', () => {
        wrapper = shallow(<TopBar {...props} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should clear local storage on logout', () => {
        wrapper = shallow(<TopBar {...props} />);
        wrapper.find('WithStyles(Button)').simulate('click');
        expect(clearSpy).toHaveBeenCalled();
        expect(window.location.reload).toHaveBeenCalled();
    });
});
