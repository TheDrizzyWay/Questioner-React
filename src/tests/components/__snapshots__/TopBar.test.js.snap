// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar /> Should maintain existing snapshot 1`] = `ShallowWrapper {}`;
